packages/block-editor/src/hooks/duotone.js

```diff
@@ -126,30 +127,59 @@ function DuotonePanel( { attributes, setAttributes } ) {
 		: duotoneStyle;
 
 	return (
-		<BlockControls group="block" __experimentalShareWithChildBlocks>
-			<DuotoneControl
-				duotonePalette={ duotonePalette }
-				colorPalette={ colorPalette }
-				disableCustomDuotone={ disableCustomDuotone }
-				disableCustomColors={ disableCustomColors }
-				value={ duotonePresetOrColors }
-				onChange={ ( newDuotone ) => {
-					const maybePreset = getDuotonePresetFromColors(
-						newDuotone,
-						duotonePalette
-					);
-
-					const newStyle = {
-						...style,
-						color: {
-							...style?.color,
-							duotone: maybePreset ?? newDuotone, // use preset or fallback to custom colors.
-						},
-					};
-					setAttributes( { style: newStyle } );
-				} }
-			/>
-		</BlockControls>
+		<>
+			<InspectorControls
+				group="filter"
+				__experimentalShareWithChildBlocks
+			>
+				<DuotoneControl
+					duotonePalette={ duotonePalette }
```

Comment:
I was thinking about what's the best way to show the origin of the duotone palettes and I'm thinking of some options here:
1. Add a new attribute to each preset to know the origin and then in the `DuotonePicker` component separate by origin if it exists. Each duotone palette would be something like `{ colors: [ '#000000', '#ffffff' ], name: 'Grayscale', slug: 'grayscale', origin: 'theme' }`. 
2. Change the shape of the `duotonePalette` object to group up by origin instead and refactor the code where we are invoking the picker so we are providing the presets already grouped. This will look cleaner and some code that we are already using to group up all the palettes can be removed, but it changes the shape of the prop significantly, and I'm not sure if this is something we should do or not. Can this component be called by third party code? We could maybe support both ways if we should, but I don't want to over complicate this.

One question here is deciding where to insert the unset duotone button if we don't really know if we will have a theme group of duotones or not (that's where it shows on the design mockups).

What are your thoughts, @ajlende ?

Reply:
I like the second option too. The component is exported publicly as `__experimentalDuotoneControl`, so it can technically be used by third-party code. I think you could add a new prop in this case, maybe something like `duotonePaletteByOrigin`, that is used instead of `duotonePalette` if it's set.